server/services/return-of-service.js

```diff
@@ -99,8 +99,45 @@ const getReturnOfServiceStatuses = async ({ participantId }) =>
       }
     );
 
+const updateReturnOfServiceStatus = async ({ participantId, newSite, newDate, newStartDate }) => {
+  const rosParticipants = await dbClient.db[collections.ROS_STATUS].find({
+    participant_id: participantId,
+    is_current: true,
+  });
+  if (rosParticipants.length === 0) {
+    throw new Error(rosError.participantNotFound);
+  }
+  const rosParticipant = rosParticipants[0];
+  if (!rosParticipant.data) {
+    throw new Error(rosError.fieldNotFound);
+  }
+
+  if (newSite && !rosParticipant.site_id) {
+    throw new Error(rosError.noSiteAttached);
+  }
+  if (newDate && !rosParticipant.data.date) {
+    throw new Error(rosError.noDate);
+  }
+  if (newStartDate && !rosParticipant.data.startDate) {
+    throw new Error(rosError.noStartDate);
+  }
+
+  dbClient.db[collections.ROS_STATUS].update(
```

Comment:
Did we talk about editing a status vs adding a new one and setting the current to `current=false`? On 
_transaction_ table I'd be wary of editing records and I'd opt for adding a new one and making it the current

If we save the editing user's ID this might even prevent the need for an audit table because we're not removing any data